projects/drivers/network/network.c

```diff
@@ -0,0 +1,502 @@
+		p = new_buffer();
+		*p_fifo = p;
+	} else {
+		p = get_last(*p_fifo);
+		p->next = new_buffer();
+		p = p->next;
+	}
+	p->instance_id = id;
+	p->data = malloc(ps->len);
+	memcpy(p->data, ps->payload, ps->len);
+	p->len = ps->len;
+}
+
+/***************************************************************************//**
+ * @brief remove_head
+*******************************************************************************/
+struct fifo * remove_head(struct fifo *p_fifo)
+{
+	struct fifo *p = p_fifo;
+	if(p_fifo != NULL) {
+		p_fifo = p_fifo->next;
+		free(p->data);
+		p->len = 0;
+		p->next = NULL;
+		p->data = NULL;
+		free(p);
+		p = NULL;
+	}
+	return p_fifo;
+}
+
+err_t network_accept(void *arg, struct tcp_pcb *newpcb, err_t err);
+err_t network_recv(void *arg, struct tcp_pcb *tpcb, struct pbuf *p, err_t err);
+void network_error(void *arg, err_t err);
+void network_store(struct tcp_pcb *tpcb, struct network_instance *es);
+void network_close(struct tcp_pcb *tpcb, struct network_instance *es);
+err_t network_poll(void *arg, struct tcp_pcb *tpcb);
+
+/***************************************************************************//**
+ * @brief network_keep_alive
+*******************************************************************************/
+void network_keep_alive(void)
+{
+	lwip_keep_alive();
+}
+
+/***************************************************************************//**
+ * @brief network_init
+*******************************************************************************/
+int32_t network_init(void)
+{
+	return init_lwip();
+}
+
+/***************************************************************************//**
+ * @brief start_application
+*******************************************************************************/
+void network_start(void)
+{
+	struct tcp_pcb *pcb;
+	u16_t port = 30431;
+	pcb = tcp_new();
+	if (pcb != NULL) {
+		err_t err;
+
+		err = tcp_bind(pcb, IP_ADDR_ANY, port);
+		if (err == ERR_OK) {
+			pcb = tcp_listen(pcb);
+			tcp_accept(pcb, network_accept);
+			xil_printf("tinyiiod server started @ port %d\n\r", port);
+		} else {
+			/* abort? output diagnostic? */
+		}
+	} else {
+		/* abort? output diagnostic? */
+	}
+}
+
+/***************************************************************************//**
+ * @brief network_accept
+*******************************************************************************/
+err_t network_accept(void *arg, struct tcp_pcb *newpcb, err_t err)
+{
+	err_t ret_err;
+	struct network_instance *es;
+	static int32_t inst_id = 0;
+	inst_id++;
+
+	LWIP_UNUSED_ARG(arg);
+	LWIP_UNUSED_ARG(err);
+
+	/* commonly observed practive to call tcp_setprio(), why? */
+	//tcp_setprio(newpcb, TCP_PRIO_MIN);
+	tcp_setprio(newpcb, TCP_PRIO_MAX);
+
+	es = (struct network_instance *)mem_malloc(sizeof(struct network_instance));
+	if (es != NULL) {
+		es->state = ES_ACCEPTED;
+		es->pcb = newpcb;
+		es->retries = 0;
+		es->p = NULL;
+		es->instance_id = inst_id;
+		HASH_ADD_INT( instances, instance_id, es );
```

Comment:
according to cppcheck there is an un-initialized variable here and in other places;
should we check if the generated code is ok ?
when following the macros, i am bit vague about it, whether the reported error is valid or not

Reply:
this looks fine; maybe here, cppcheck has a limitation;
i suggest using inline suppression comments;
so, if you add `/* cppcheck-suppress uninitvar ;  cppcheck reports this as a false positive */` , cppcheck should consider the comment and ignore this

Reply:
related to comment https://github.com/analogdevicesinc/no-OS/pull/143/files#r257196836   please take a look at this comment;
you can try 
```
/* cppcheck-suppress uninitvar ; cppcheck reports this as a false positive */
```
which avoids cppcheck going over this line only